app/src/main/java/org/listenbrainz/android/ui/screens/listens/ListensScreen.kt

```diff
@@ -145,12 +190,73 @@ fun ListensScreen(
                     coverArtUrl = Utils.getCoverArtUrl(
                         caaReleaseMbid = listen.trackMetadata.mbidMapping?.caaReleaseMbid,
                         caaId = listen.trackMetadata.mbidMapping?.caaId
-                    )
+                    ),
+                    dropDown = {
+                        SocialDropdown(
+                            isExpanded = dropdownItemIndex.value == index,
+                            onDismiss = {
+                                dropdownItemIndex.value = null
+                            },
+                            metadata = metadata,
+                            onRecommend = {
+                                try {
+                                    socialViewModel.recommend(metadata)
```

Comment:
This is wrong, we do not propagate errors to the UI right now so this will always result in a successful toast. If you want, you can propagate the error via `Resource` class and then check if the response was successful then return error. We can ditch errorFlow for this as well.

Reply:
Remember its okay to change the core code if you feel its right. At most I will say is "This is wrong" and you will learn a thing or two. Great work overall.